packages/frontend/src/components/common/Filters/renderFilterItem.tsx

```diff
@@ -17,17 +48,20 @@ export const renderFilterItem: ItemRenderer<Field | TableCalculation> = (
         return null;
     }
     return (
-        <MenuItem2
-            key={getItemId(item)}
-            roleStructure="listoption"
-            shouldDismissPopover={false}
-            active={modifiers.active}
-            disabled={modifiers.disabled}
-            icon={<FieldIcon item={item} />}
-            text={<FieldLabel item={item} />}
-            onClick={handleClick}
-            onFocus={handleFocus}
-        />
+        <Tooltip
+            withinPortal
+            label={isField(item) ? item.description : ''}
+            disabled={!isField(item) || (isField(item) && !item.description)}
+        >
+            <FilterItem
+                item={item}
+                active={modifiers.active}
+                disabled={modifiers.disabled}
+                text={<FieldLabel item={item} />}
```

Comment:
Why is the `text` set here but the `icon` is set inside `FilterItem` ? I haven't worked with these components so maybe it's obvious to frontend devs but I was surprised

Reply:
oh, good q, I set the `text` here because it's different between `renderFilterItem` and `renderFilterItemWithoutTableName`, but the `icon` is the same, so let  `FilterItem` set it there. 
This is not reused anywhere else so did it this way 😅